Add Astro CLI <> Platform Versioning guidelines to CLI Quickstart + Versioning Matrix (#225)
astronomer/docs: enterprise/v0.23/02_develop/01_cli-quickstart.md

$ astro dev start
```

## Astronomer CLI and Platform Versioning

> For every minor version of the Astronomer Platform, a corresponding minor version of the Astronomer CLI is made available. To ensure reliability, upgrade to the corresponding minor version of the Astronomer CLI. If you're on Astronomer v0.23+, for example, Astronomer CLI v0.23+ is required.

Reviewer: ```suggestion
For every minor version of the Astronomer platform, a corresponding minor version of the Astronomer CLI is made available. To stay up to date, upgrade to the corresponding minor version of the Astronomer CLI whenever you upgrade your Astronomer platform. If you're on Astronomer v0.23+, for example, Astronomer CLI v0.23+ is required.
```

Reviewer: It sounds like having an out-of-date CLI can cause breakages based on the potential error message mentioned below. I wanted to use firmer language because of that, but seeing as this is in the quickstart I can see why we'd want to be a little more welcoming. Also doing this won't necessarily "ensure" a frictionless experience on its own, so I'd move away from that language.

What about: "To deploy successfully, you should always upgrade to the corresponding minor version of the Astronomer CLI when you upgrade to a new minor version of Astronomer." Any variation on that is cool with me!

> While corresponding minor versions of the Astronomer Platform and the Astronomer CLI are required, subsequent patch versions do not need to match and will not introduce breaking changes. The release of Astronomer v0.23.9, for example, does not mean that v0.23.9 of the Astronomer CLI exists or is required.

Reviewer: ```suggestion
While upgrading to a minor version of Astronomer requires upgrading the Astronomer CLI, subsequent patch versions do not need to match and will not introduce breaking changes. For instance, upgrading from Astronomer v0.23.0 to v0.23.9 would not require upgrading the Astronomer CLI.
```

Author: What about:

```suggestion
While upgrading to a new minor version of Astronomer requires to a corresponding minor version of the Astronomer CLI, subsequent patch versions of each will be released on an ad-hoc basis and not necessarily map to each other directly. For instance, v0.23.2 may be the latest available version of the Astronomer CLI, while v0.23.9 is the latest available version of Astronomer. While we encourage users to always run the latest version of all components, patch versions of both Astronomer and the Astronomer CLI within the v0.23 series will always be compatible with one another and will not introduce breaking changes.
```

Reviewer: Cleaning this up a bit:
```suggestion
While upgrading to a new minor version of Astronomer requires upgrading the Astronomer CLI, subsequent patch versions of each product will remain compatible. For instance, consider a system where Astronomer is on v0.23.9 and the Astronomer CLI is on v0.23.2. While we encourage users to always run the latest version of all components, patch versions of both Astronomer and the Astronomer CLI within the v0.23 series will always be compatible with each other such that these versions would be compatible with each other. 
```
I like the use of "series" here!

### Check Running Astronomer CLI and Platform Versions

> To check your working version of the Astronomer CLI (`Astro CLI`) and the Astronomer Platform (`Astro Server Version`), run:

Reviewer: ```suggestion
To check your working versions of Astronomer (`Astro Server Version`) and the Astronomer CLI (`Astro CLI`), run:
```

> ```

Reviewer: ```suggestion
```sh
```

$ astro version
```

This command will output something like the following:

```sh
$ astro version

> Astro CLI Version: 0.23.2

Reviewer: ```suggestion
Astro CLI Version: 0.23.9
```
Since we don't show the error message here, should we have this example be something that wouldn't give the error?

Author: @jwitz This wouldn't show an error! The whole point is that you have to be on `0.23.x` of both, but the two won't map 1:1.

The latest version of the Astronomer CLI = v0.23.2
The latest version of Astronomer = v0.23.9

They are compatible with one another - v0.23.9 of the Astro CLI does not exist and will never necessarily exist

Reviewer: Ah yep my bad! Confused minor and patch here.

Astro Server Version: 0.23.9
Git Commit: 748ca2e9de1e51e9f48f9d85eb8315b023debc2f
```

> If the minor versions between the two don't match, you'll receive an error message in your command line with instructions to either upgrade or downgrade. If you have access to more than one Astronomer Enterprise installation, `Astro Server Version` will correspond to the `<base-domain>` that you're currently authenticated into.

Reviewer: ```suggestion
If the minor versions between the two components don't match, you'll receive an error message in your command line with instructions to either upgrade or downgrade. If you have access to more than one Astronomer Enterprise installation, `Astro Server Version` will correspond to the `<base-domain>` that you're currently authenticated into.
```

> For more information on Astronomer Platform and CLI releases, refer to:

Reviewer: ```suggestion
For more information on Astronomer and Astronomer CLI releases, refer to:
```
